src/command_modules/azure-cli-network/azure/cli/command_modules/network/tests/test_network_commands.py

```diff
@@ -634,6 +634,31 @@ def body(self):
         self.cmd('network nic ip-config update -g {} --nic-name {} -n {} --subnet {} --vnet-name {}'.format(rg, nic, config, subnet, vnet),
             checks=JMESPathCheck("subnet.contains(id, '{}')".format(subnet), True))
 
+class NetworkNicConvenienceCommandsScenarioTest(ResourceGroupVCRTestBase):
+
+    def __init__(self, test_method):
+        super(NetworkNicConvenienceCommandsScenarioTest, self).__init__(__file__, test_method)
+        self.resource_group = 'cli_nic_convenience_test'
+        self.vm_name = 'conveniencevm1'
+
+    def test_network_nic_convenience_commands(self):
+        self.execute()
+
+    def set_up(self):
+        super(NetworkNicConvenienceCommandsScenarioTest, self).set_up()
+        rg = self.resource_group
+        vm = self.vm_name
+        self.cmd('vm create -g {} -n {} --image UbuntuLTS --admin-password aBcD1234!@#$ --authentication-type password'.format(rg, vm))
+
+    def body(self):
+        rg = self.resource_group
+        vm = self.vm_name
+        nic_id = self.cmd('vm show -g {} -n {} --query "networkProfile.networkInterfaces[0].id"'.format(rg, vm))
+        self.cmd('network nic list-effective-nsg --ids {}'.format(nic_id),
+            checks=JMESPathCheck("length(value)", 1))
+        self.cmd('network nic show-effective-route-table --ids {}'.format(nic_id),
+            checks=JMESPathCheck("length(value)", 7))
```

Comment:
Actually I still never could get JMESPath to play nice here. I used a plain ol' assert.

---
In reply to: [79085537](https://github.com/Azure/azure-cli/pull/903#discussion_r79085537) [](ancestors = 79085537)